index.js

```diff
@@ -19,28 +19,20 @@ module.exports = class extends Filter {
 		return 'css';
 	}
 
-	processString(text, relativePath) {
-		const options = Object.assign({
+	async processString(text, relativePath) {
+		const options = {
 			from: relativePath,
-			to: relativePath
-		}, this.options);
+			to: relativePath,
+			...this.options
+		};
 
 		// Support explicit override of inline sourcemaps
 		if (options.sourcemap !== null || options.sourcemap !== undefined) {
 			options.map = options.sourcemap ? 'inline' : false;
 		}
 
-		return postcss(autoprefixer(options))
+		const result = await postcss(autoprefixer(options))
 			.process(text, options)
-			.then(result => {
-				const warnings = result.warnings();
-
-				if (warnings.length > 0) {
-					console.error(warnings.join('\n'));
-				}
-
-				return result.css;
-			})
 			.catch(error => {
```

Comment:
should we mix `await` and `.catch` or should this use `await` and `try/catch`?

Reply:
I was under the impression that try/catch was a poor pattern due to deoptimizing in chrome, but my knowledge may be out of date :-\

Reply:
@nlfurniss you are correct that used to be true, as part of the TurboFan effort it now optimizes as expected. Bonus, when using async/await with try/catch it tends to be faster then the explicit promise variant, and more importantly it provides better zero-cast async stack-traces.